homeassistant/components/aprilaire/climate.py

```diff
@@ -158,8 +159,14 @@ def hvac_action(self) -> HVACAction | None:
     @property
     def current_temperature(self) -> float | None:
         """Get current temperature."""
-        return self.coordinator.data.get(
-            Attribute.INDOOR_TEMPERATURE_CONTROLLING_SENSOR_VALUE
+        return convert_temperature_if_needed(
+            self.hass.config.units.temperature_unit,
+            cast(
+                float,
+                self.coordinator.data.get(
+                    Attribute.INDOOR_TEMPERATURE_CONTROLLING_SENSOR_VALUE
+                ),
+            ),
```

Comment:
convert_temperature_if_needed doesn't accept a None if you're confident your `self.coordinator.data` is present you can use `self.coordinator.data[Attribute.INDOOR_TEMPERATURE_CONTROLLING_SENSOR_VALUE]`

since `current_temperature` can return a None, and likely will when first polling I'd do this:

```
try:
    return convert_temperature_if_needed(
        self.hass.config.units.temperature_unit, 
        self.coordinator.data[Attribute.INDOOR_TEMPERATURE_CONTROLLING_SENSOR_VALUE
    )
except KeyError:
    return None
```

if mypy complains you can explicitly check for the key with an if statement.

My 2¢, avoid casting whenever at all possible

Reply:
@EvanSchalton Thank you for your feedback. The `cast` is because `self.coordinator.data` is `dict[str, Any]`, so I am casting specifically to a `float`. Technically the data itself can be `str | float`, but in this case it is always `float`. I updated the type hint on `convert_temperature_if_needed` to be `float | None` (which was already handled).

Reply:
I see in your other comment that we said the same thing.